Gems/ScriptCanvasDeveloper/Code/Editor/Source/TSGenerateAction.cpp

```diff
@@ -6,157 +6,269 @@
+        }
 
-            AZ::BehaviorContext* behaviorContext{};
-            AZ::ComponentApplicationBus::BroadcastResult(behaviorContext, &AZ::ComponentApplicationRequests::GetBehaviorContext);
+        void ReloadTranslation(QWidget*)
+        {
+            GraphCanvas::TranslationRequestBus::Broadcast(&GraphCanvas::TranslationRequests::Restore);
+            ScriptCanvasEditor::AssetTrackerRequestBus::Broadcast(&ScriptCanvasEditor::AssetTrackerRequests::RefreshAll);
+        }
+
+        QAction* TranslationDatabaseFileAction(QMenu* mainMenu, QWidget* mainWindow)
+        {
+            QAction* qAction = nullptr;
+
+            if (mainWindow)
+            {
+                qAction = new QAction(QAction::tr("Produce Localization Files for All Types"), mainWindow);
+                qAction->setAutoRepeat(false);
+                qAction->setToolTip("Produces a .names file for every reflected type supported by scripting.");
+                qAction->setShortcut(QKeySequence(QAction::tr("Ctrl+Alt+X", "Debug|Produce Localization Database")));
+                mainWindow->addAction(qAction);
+                mainWindow->connect(qAction, &QAction::triggered, &GenerateTranslationDatabase);
+
+                qAction = mainMenu->addAction(QAction::tr("Translation Browser"));
+
+                qAction->setAutoRepeat(false);
+                qAction->setToolTip("....");
+                qAction->setShortcut(QAction::tr("Ctrl+Alt+T", "Developer|Translation Browser"));
+                QObject::connect(qAction, &QAction::triggered, [mainWindow]() {OpenTranslationBrowser(mainWindow); });
+
+                qAction = mainMenu->addAction(QAction::tr("Reload Translation"));
+                qAction->setAutoRepeat(false);
+                qAction->setToolTip("....");
+                qAction->setShortcut(QAction::tr("Ctrl+Alt+R", "Developer|Reload Translation"));
+                QObject::connect(qAction, &QAction::triggered, [mainWindow]() {ReloadTranslation(mainWindow); });
+
+            }
+
+            return qAction;
+        }
+
+        template <typename T>
+        bool ShouldSkip(const T* object)
+        {
+            using namespace AZ::Script::Attributes;
+
+            // Check for "ignore" attribute for ScriptCanvas
+            auto excludeClassAttributeData = azdynamic_cast<const AZ::Edit::AttributeData<ExcludeFlags>*>(AZ::FindAttribute(ExcludeFrom, object->m_attributes));
+            const bool excludeClass = excludeClassAttributeData && (static_cast<AZ::u64>(excludeClassAttributeData->Get(nullptr)) & static_cast<AZ::u64>(ExcludeFlags::List | ExcludeFlags::Documentation));
+
+            if (excludeClass)
+            {
+                return true; // skip this class
+            }
 
-            if (serializeContext == nullptr || behaviorContext == nullptr)
+            return false;
+        }
+
+        void WriteString(rapidjson::Value& owner, const AZStd::string& key, const AZStd::string& value, rapidjson::Document& document)
```

Comment:
There wasn't a JsonUtil.h function for this?

Reply:
This is fairly specific for what it's used, also it will be removed